instill/model/v1alpha/model.proto

```diff
@@ -4,153 +4,341 @@ package instill.model.v1alpha;
 
 // Protobuf standard
 import "google/protobuf/struct.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
 
 // Google api
-import "google/api/annotations.proto";
+import "google/api/resource.proto";
 import "google/api/field_behavior.proto";
 
 import "instill/model/v1alpha/definition.proto";
 
-// LivenessRequest represents a request to check a service liveness status
-message LivenessRequest {
-  // Service name to check for its liveness status
-  string service = 1 [ (google.api.field_behavior) = OPTIONAL ];
-}
-
-// LivenessResponse represents a response for a service liveness status
-message LivenessResponse {
-  // ServingStatus enumerates the status of a queried service
-  enum ServingStatus {
-    // Serving status: UNSPECIFIED
-    SERVING_STATUS_UNSPECIFIED = 0;
-    // Serving status: SERVING
-    SERVING_STATUS_SERVING = 1;
-    // Serving status: NOT SERVING
-    SERVING_STATUS_NOT_SERVING = 2;
+// Model represents a model
+message Model {
+  option (google.api.resource) = {
+    type: "api.instill.tech/Model"
+    pattern: "models/{model}"
+  };
+
+  // Model visibility including public or private
+  enum Visibility {
+    // Visibility: UNSPECIFIED, equivalent to PRIVATE.
+    VISIBILITY_UNSPECIFIED = 0;
+    // Visibility: PRIVATE
+    VISIBILITY_PRIVATE = 1;
+    // Visibility: PUBLIC
+    VISIBILITY_PUBLIC = 2;
   }
 
-  // Status is the instance of the enum type ServingStatus
-  ServingStatus status = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-}
-
-// ReadinessRequest represents a request to check a service readiness status
-message ReadinessRequest {
-  // Service name to check for its readiness status
-  string service = 1 [ (google.api.field_behavior) = OPTIONAL ];
+  // Resource name. It must have the format of "models/{model}".
+  // For example: "models/yolov4"
+  string name = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ] ;
+  // Model ID in UUIDv4
+  string id = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model display name, this is used to construct the resource name.
+  string display_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Model description
+  optional string description = 4 [ (google.api.field_behavior) = OPTIONAL ];
+  // Model definition UUID
+  string model_definition = 5 [ (google.api.resource_reference).type = "api.instill.tech/ModelDefinition", (google.api.field_behavior) = IMMUTABLE ];
+  // Model definition configuration
+  Spec configuration = 6 [ (google.api.field_behavior) = IMMUTABLE ];
```

Comment:
@xiaofei-du this spec is same as in model_definition right?

Reply:
Not really. They are both arbitary JSON object, but stores different values. 

Use connector-backend's Azure blob storage destination as an example.
spec in connector-definition will be [this](https://github.com/instill-ai/connector-backend/blob/88728cce8c6d73f31f7fe084e825de33eac8b9fe/configs/init/airbyte/seed/destination_specs.yaml#L8-L86). It stores the specs like what fields are need to be configurated to use this destination. Similarly, `model_definition.spec` is to store what fields need to be configured for this model definition (local or GitHub).

`model.configuration` will be the user-provided configuration values according to the `model_definition.spec`.